src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -210,4 +266,61 @@ class EntityClient (requestContext: RequestContext)(implicit protected val execu
       }
     }
   }
+
+  def importBagit(pipeline: WithTransformerConcatenation[HttpRequest, Future[HttpResponse]], workspaceNamespace: String, workspaceName: String, bagitRq: BagitImportRequest): Future[PerRequestMessage] = {
+    if(bagitRq.format != "TSV") {
+      Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "Invalid format; for now, you must place the string \"TSV\" here"))
+    } else {
+
+      //Java URL handles http, https, ftp, file, and jar protocols.
+      //We're only OK with https to avoid MITM attacks.
+      val bagitURL = new URL(bagitRq.bagitURL)
+      val acceptableProtocols = Seq("https") //for when we inevitably change our mind and need to support others
+      if (!acceptableProtocols.contains(bagitURL.getProtocol)) {
+        Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "Invalid bagitURL protocol: must be https only"))
+      } else {
+
+        val rand = java.util.UUID.randomUUID.toString.take(8)
+        val bagItFile = File.createTempFile(s"$rand-samples", ".tsv")
+
+        try {
+          val conn = bagitURL.openConnection()
```

Comment:
does this connection need to be closed?

Reply:
URLConnection doesn't seem to have a disconnect or close method. Reading on the internet, it appears you only have to close it if you open a stream? I _think_ it's okay like this (closes on its own).

Reply:
One thing you could do is cast it as a `HttpURLConnection` and use that class' disconnect. Take this with a grain of salt as I'm not a huge fan of casting.